packages/babel-generator/src/source-map.ts

```diff
@@ -80,20 +106,46 @@ export default class SourceMap {
   ) {
     this._rawMappings = undefined;
 
+    let original;
+    let source;
+
+    if (line != null) {
+      if (this._inputMap) {
+        const originalMapping = originalPositionFor(this._inputMap, {
+          line,
+          column,
+        });
+
+        source =
+          originalMapping.source ||
+          filename?.replace(/\\/g, "/") ||
+          this._sourceFileName;
+
+        if (originalMapping.line != null) {
+          original = {
+            line: originalMapping.line,
+            column: originalMapping.column,
+          };
+          if (originalMapping.name) {
+            identifierName = originalMapping.name;
```

Comment:
With this, I don't think the `originalPositionFor` in the printer are necessary.